charts/fwd-proxy/chart/values.yaml

```
@@ -42,6 +43,91 @@ nginx:
     #   cpu: 100m
     #   memory: 128Mi
 
+caddy:
+  enabled: true
+  image:
+    repository: caddy
+    pullPolicy: IfNotPresent
+    # Overrides the image tag whose default is the chart appVersion.
+    tag: "2.5.2-alpine"
+
+  securityContext: {}
+    # capabilities:
+    #   drop:
+    #   - ALL
+    # readOnlyRootFilesystem: true
+    # runAsNonRoot: true
+    # runAsUser: 1000
+
+  resources: {}
+    # We usually recommend not to specify default resources and to leave this as a conscious
+    # choice for the user. This also increases chances charts run on environments with little
+    # resources, such as Minikube. If you do want to specify resources, uncomment the following
+    # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
+    # limits:
+    #   cpu: 100m
+    #   memory: 128Mi
+    # requests:
+    #   cpu: 100m
+    #   memory: 128Mi
+
+  caddyfile: |
+    {
+        debug
+        auto_https disable_redirects
+        local_certs
+        skip_install_trust
+    }
+
+    :80 {
+      respond / 404
+      respond /healthz 200
+    }
+
+    {{- $registries := list "registry-1.docker.io" "ghcr.io" "k8s.gcr.io" "quay.io" "gridopsingularity.azurecr.io" }}
+    {{- range $registries }}
+    {{ . }}:443 {
+      respond / 200
+
+      reverse_proxy /v2/ http://image-registry-service
+      reverse_proxy /v2/_catalog http://image-registry-service
+
+      @pathAfterVersion {
+        path_regexp xp ^/v2/(.+)
+      }
+
+      {{- if eq . "registry-1.docker.io" }}
```

Let's keep the comment from the nginx config:
```suggestion
      {{- /* docker.io is mapped to registry-1.docker.io in the code:
           * https://github.com/containers/image/blob/b55fb86c28b7d743cf59701332cd78d4294c7c54/docker/docker_client.go#L249-L251
           */}}
      {{- if eq . "registry-1.docker.io" }}
```